### DISCUSSION TITLE
Export data to Excel

### DISCUSSION THREAD

**Discussion by Manishkparmar80** [Budibase/budibase#1885] [Ideas]
How to export data to excel. 
![Export to Excel](https://user-images.githubusercontent.com/84949590/124001467-4ce01900-d9f2-11eb-9e44-f5aeea0c04c0.JPG)

**Comment by oswaldB** [comment]
Hello,

I don't think budibase can do it right now. But if you must use an open source lowcode software you can install n8n and make a connection between budibase and n8n using webhook.

2 possibilities:
1- you have an internal database in budi. Then you have to send the data to your n8n webhook node and then use the spritesheet node and the export to binary node to have your excel.
2 - you use an external database in budi. Then you have to send the webhook to n8n. Then in n8n you do your request using a node of the technology of your database and then the spreadsheet node and then the binary node.

I have never made it but it is the way I'll do this.

Please tell me if it works ;)

**Reply by Manishkparmar80** [reply]
@oswaldB Thanks.

**Reply by chadgriff** [reply]
Hey @oswaldB,

Just wanted to let you know I was able to setup n8n and successfully write an .xls file to local disk based on the #2 scenario you provided.  Thanks for the idea!

**Comment by mjashanks** [comment]
There is no in-built way to export data from a Budibase app. 

You can export data from tables in the `Data` section of the builder: 

![image](https://user-images.githubusercontent.com/3524181/124100064-0a502800-da56-11eb-9a0a-8f25183abea9.png)

I've changed this discussion to an "Idea", as it seems like a common requirement.

**Reply by Manishkparmar80** [reply]
@mjashanks Hi, i am trying to export data to excel through published application so direct export data from tables will not be suitable.

**Comment by mjashanks** [comment]
Also requested here: https://github.com/Budibase/budibase/discussions/2907

**Comment by rhrish** [comment]
An alternate approach I've been using is by writing a google sheets App script that reads from the database using JDBC Connector API (you need to use an external DB like MySQL/Postgres and your database should be accessible from the internet) and populate a google spreadsheet. It involves a little bit of coding though.

**Comment by tluyben** [comment]
This would be a nice feature; it would have to respect be able to respect filters within that block like the view should, but then it would be great for enterprise adoption as you need export to xxx for everything in those envs. 

We would typically not be allowed to 'send data to y, and export z' due to the nature of the data anyway.

A transformation JS option to the export would be nice as well so you can make formats that no one needs, like obscure banking stuff we always need.

**Comment by Rory-Lambert** [comment]
Hello! I know this is an old thread. Has anything like this been implemented yet? Would be very useful for my application. This is one of the top google results on the topic!

**Reply by ConorWebb96** [reply]
You can export data in Budibase to CSV and then import the CSV file into Excel. The relevant documentation for this can be found [here](https://docs.budibase.com/docs/data-actions#export-data).

**Reply by Rory-Lambert** [reply]
This is exactly what I was after - thanks Conor. Somehow missed the official docs, but was very easy after you pointed them out. Cheers!
